### DISCUSSION TITLE
"The connection string ... has no supporting driver"

### DISCUSSION THREAD

**Discussion by roberte777** [SeaQL/sea-orm#514] [Q&A] [answered]
Hey there, I've had a lot of experience using ORM's in Javascript, this is my first step into it with Rust, so forgive me if I don't know several rust specific things. When trying to initialize my database with some starting migrations, I get the error: 
The connection string 'postgresql://user:pass@localhost:5432/db has no supporting driver.
I took out any specific info relating to my implementation from the string. Any ideas why this would be happening? Thanks for any help!

**Comment by billy1624** [comment, marked as answer]
Hey @roberte777, welcome to Rust!

You need to enable the database feature and runtime.

https://www.sea-ql.org/SeaORM/docs/install-and-config/database-and-async-runtime

**Reply by roberte777** [reply]
![image](https://user-images.githubusercontent.com/33569440/153638251-ccf128e3-5f5a-41d5-ac76-56fe87ea7024.png)
never uploaded an image from clipboard to github, so hopefully this works. I had this in my main Cargo.toml, something in here not correct?

**Reply by billy1624** [reply]
This is correct!

However, you need to structure the cargo workspace by following the steps below.

https://www.sea-ql.org/SeaORM/docs/migration/setting-up-migration#workspace-structure

**Reply by billy1624** [reply]
Gonna sleep now loll
Talk to you tmr

**Reply by roberte777** [reply]
Cool, I didn't see specifically in the docs that it says to list database driver in the toml file for the entity folder. Just thought it was the same as the core toml command. For documentation purpose, it could be helpful if small, easy to miss changes like that between locations were mentioned specifically, making them harder to miss. That aspect is working now, thank you!

**Reply by billy1624** [reply]
Good suggestions! Definitely have room for improvements. You can also click on the `Edit this page` button to propose changes! :P

<img width="400" alt="image" src="https://user-images.githubusercontent.com/30400950/153695450-263d6d04-9958-4213-b2a3-01d8807fbf2b.png">

**Comment by CPoncet** [comment]
I can attest to the lack of precision in the docs. It took me a while to figure out I had to adjust Cargo.toml in the migration folder to specify runtime and db driver. I thought it was enough to stipulate it in the main crate's Cargo.toml. It worked perfectly afterwards.

**Reply by billy1624** [reply]
Hey @CPoncet, thanks for raising this issue up! We've added some comments on the generated migration crate (`sea-orm-cli migrate init`)

https://github.com/SeaQL/sea-orm/blob/bb8a8bc286f2788496e0712d45ab42cc6a7093d2/sea-orm-cli/template/migration/_Cargo.toml#L14-L22

However, as you've said we didn't mention it specifically on the documentation site. I guess we can add these comments to this section: https://www.sea-ql.org/SeaORM/docs/migration/setting-up-migration/#workspace-structure

```toml
[dependencies]
async-std = { version = "^1", features = ["attributes", "tokio1"] }
sea-orm-migration = { version = "^0" }
```

Would that be enough?

**Reply by CPoncet** [reply]
I figured out about the comments on the generated migration crate, but only when I read this discussion. 

So yeah adding it in the docs is way better in my opinion.

**Reply by billy1624** [reply]
Added! https://github.com/SeaQL/seaql.github.io/commit/2792e497c6799425db8bc0ad0da232333ffe3d0d
